subsys/bluetooth/host/smp.c

```diff
@@ -3928,6 +3949,48 @@ static uint8_t smp_ident_info(struct bt_smp *smp, struct net_buf *buf)
 	return 0;
 }
 
+static uint8_t smp_id_add_unpair(struct bt_smp *smp, struct bt_keys *new_bond)
+{
+	struct bt_keys * conflict;
+
+	/* Sanity check: It does not make sense to finalize a bond before we
+	 * have the remote identity.
+	 */
+	__ASSERT_NO_MSG(!(smp->remote_dist & BT_SMP_DIST_ID_KEY));
+
+	conflict = bt_id_find_conflict(new_bond);
+
+	if (conflict) {
+		BT_INFO("New bond conflicts with a bond on id %d.", conflict->id);
+	}
+
+	if (conflict && !IS_ENABLED(BT_ID_UNPAIR_MATCHING_BONDS)){
+		BT_WARN("Refusing new pairing. The old bond must be unpaired first.");
+		return BT_SMP_ERR_AUTH_REQUIREMENTS;
+	}
+
+	if (conflict && IS_ENABLED(BT_ID_UNPAIR_MATCHING_BONDS)) {
```

Comment:
Are you sure you're exercising this branch in your test? I think it needs the `CONFIG_` prefix, I just used `IS_ENABLED` recently.